openedx/core/djangoapps/credit/signals.py

```diff
@@ -80,12 +79,38 @@ def listen_for_grade_calculation(sender, username, grade_summary, course_key, de
             criteria = requirements[0].get('criteria')
             if criteria:
                 min_grade = criteria.get('min_grade')
-                if grade_summary['percent'] >= min_grade:
-                    reason_dict = {'final_grade': grade_summary['percent']}
-                    api.set_credit_requirement_status(
-                        username, course_id, 'grade', 'grade', status="satisfied", reason=reason_dict
-                    )
-                elif deadline and deadline < timezone.now():
+                now = timezone.now()
+                status = None
+                reason = None
+
+                if (deadline and now < deadline) or not deadline:
+                    # Student completed coursework on-time
+
+                    if grade_summary['percent'] >= min_grade:
```

Comment:
Acknowledged. Will fix in a future visit to this codebase. As much as I hope that never happens...it'll probably be in about 3 months! 😭